src/compiler/checker.ts

```diff
@@ -13014,7 +13025,9 @@ namespace ts {
                         getCombinedMappedTypeOptionality(source) <= getCombinedMappedTypeOptionality(target));
                 if (modifiersRelated) {
                     let result: Ternary;
-                    if (result = isRelatedTo(getConstraintTypeFromMappedType(target), getConstraintTypeFromMappedType(source), reportErrors)) {

```

Comment:
Is it possible to only report as unmeasurable when the mapped types have modifiers?

Reply:
Isn't the `Record` example exactly the case of when there isn't a modifier?

Reply:
Yep, adding the modifier check means that this will not fix #29698. But the two issues aren't exactly the same so I was wondering if there exists two distinct fixes. #29698 only really breaks because when using `string` as a key type.

Reply:
This was the kind of (**hack**) I had in mind: https://pastebin.com/n431Puhu. I have no idea if this is even feasible, or correct. I thought I'd at least play around with something rather than solely firing off random questions.